graphql-jpa-query-schema/src/main/java/com/introproventures/graphql/jpa/query/schema/impl/IntrospectionUtils.java

```diff
@@ -24,25 +33,38 @@ public static CachedIntrospectionResult introspect(Class<?> entity) {
     }
 
     public static boolean isTransient(Class<?> entity, String propertyName) {
-        return isAnnotationPresent(entity, propertyName, Transient.class);
+        if(!introspect(entity).hasPropertyDescriptor(propertyName)) {
+            throw new RuntimeException(new NoSuchFieldException(propertyName));
+        }
+        
+        return Stream.of(isAnnotationPresent(entity, propertyName, Transient.class),
+                         isModifierPresent(entity, propertyName, Modifier::isTransient))
+                     .anyMatch(it -> it.isPresent() && it.get() == true);
     }
-
+    
     public static boolean isIgnored(Class<?> entity, String propertyName) {
-        return isAnnotationPresent(entity, propertyName, GraphQLIgnore.class);
+        return isAnnotationPresent(entity, propertyName, GraphQLIgnore.class)
+                 .orElseThrow(() -> new RuntimeException(new NoSuchFieldException(propertyName)));
     }
 
-    private static boolean isAnnotationPresent(Class<?> entity, String propertyName, Class<? extends Annotation> annotation){
+    private static Optional<Boolean> isAnnotationPresent(Class<?> entity, String propertyName, Class<? extends Annotation> annotation){

```

Comment:
Returning `Optional<Boolean>` is a little bit misleading in this case. In my opinion calling isXXX methods should always return pure boolean value by convention. Using API with `Optional<Boolean>` leads us to three-state logic we need to deal with. What do you think?